README.md

```diff
@@ -1,2 +1,166 @@
+    plugins in the same browser tab.
+  - Flexible, custimizable tabbed and paneled layout system.
+  - Initial plugins:
+    - Menu bar.
+    - Overall page layout.
+    - File browser.
+    - Text editor.
+    - Terminal.
+* All new frontend work is being done as small, loosely coupled npm packages.
+* The entire Jupyter Workbench will be a set of plugins (again just npm packages)
+  that exchange runtime APIs.
+* Application infrastucture (keyboard shortcuts, commands, menus, layout), view
+  layer, and other utilities (signals, properties, messages) provided by phosphorjs
+  (http://phosphorjs.github.io/).
+* Here is a list of all the repositories in which we are developing the Jupyter Workbench:
+  - https://github.com/jupyter?utf8=%E2%9C%93&query=jupyter-js
+* Movement towards having stable, documented JavaScript APIs.
+* Usability and UX User survey to help us guide the design process of the Jupyter Workbench.
+* Ability to hide/show code and entire cells based on query syntax.
+
+### Next+2
+
+* New plugins for the Jupyter Workbench:
+  - Standalone output area.
+  - Ability to hook kernels up to text editors and output areas.
+  - Themes.
+
+### Future
+
+* New plugins for the Jupyter Workbench:
+  - Notebook.
+  - Variable inspector.
+  - UI for managing plugins.
+* Dashboarding of output areas and interactive widgets.
+* As, and only if, the Jupyter Workbench becomes a full replacement, deprecate existing pages in the notebook
+  web-application.
+* Real-time collaboration on the notebook, text editor, and other plugins.
+* Perform and publish accesibility audit by running an automated tool.
+* Bring our core plugins up to the level of the Web Accessibility Standards
+  (http://www.w3.org/standards/webdesign/accessibility)
+
+## ipywidgets
+
+### Next (5.0)
+
+* Improved and consistent visual style of built-in widgets.
+* Better abstractions for styling widgets using CSS and doing layout with flexbox.
+* Ability to run ipywidgets outside of the notebook.
+* Initial refactoring to start to use npm for packaging the JavaScript code.
+
+### Future
+
+* Further decouple the JavaScript and Python kernel backend, likely moving the JavaScript
+  out of the Python repo into the Jupyter org.
+* Explore how we can start to treat widgets as regular output, and not have to have
+  a separate widget area in the DOM.
+* Explore how the existing codebase will integrate with phosphor. In particular, explore
+  using phosphor properties to abstract away the details of backbone.js models and replace
+  the backbone models with phosphorjs widget-based views.
+* Transition to ES6/TypeScript.
+* Allow widgets to be used at the top-level of the Jupyter Workbench UI to create
+  dashboards.
+
+## traitlets
+
+### Next (4.1)
+
+* Much nicer, decorator based APIs for observing (`@observe`), validating
+  (`@validate) and providing defaults (`@default) for traits.
+* Looking forward to its usage in Matplotlib.
+
+## nbconvert
+
```

Comment:
### Next

* Bring examples of functionality that exists in locations like @takluyver's demos from last summer's Southampton workshop into the official docs.